docs/workflow.md

```diff
@@ -38,12 +38,8 @@ In the event that spawning the workflow would result in recursion, the child wor
 will be marked as failed with a message explaining that recursion was detected.
 This is to prevent saturation of the task system with an infinite chain of workflows.
 
-### Tree-Graph Formation and Restrictions
```

Comment:
Line 4:

> A workflow has an associated tree-graph that is composed of multiple nodes.

It is no longer a tree-graph.

I'm editing this file in `workflow_inventory` branch now, after @matburt's PR review of it.

I hope we can avoid merge conflicts, but we may have some.